File: 3/list-3.go
package main

import (
	"errors"
	"fmt"
)

type ContaBancaria struct {
	nome string
	saldo   float64
}

func (c *ContaBancaria) Depositar(valor float64) {
	if valor <= 0 {
		fmt.Println("O valor do depósito deve ser positivo.")
		return
	}
	c.saldo += valor
	fmt.Printf("Depositado: R$%.2f\n", valor)
}

func (c *ContaBancaria) Sacar(valor float64) error {
	if valor <= 0 {
		return errors.New("o valor de saque deve ser positivo")
	}
	if valor > c.saldo {
		return errors.New("saldo insuficiente")
	}
	c.saldo -= valor
	fmt.Printf("Sacado: R$%.2f\n", valor)
	return nil
}

func (c *ContaBancaria) Saldo() float64 {
	return c.saldo
}

func main() {
	conta := ContaBancaria{"Marcio", 1000.0} 

	fmt.Printf("Saldo inicial de %s: R$%.2f\n", conta.nome, conta.Saldo())

	conta.Depositar(500)
	fmt.Printf("Saldo após depósito: R$%.2f\n", conta.Saldo())

	if err := conta.Sacar(200); err != nil {
		fmt.Println("Erro ao sacar:", err)
	} else {
		fmt.Printf("Saldo após saque: R$%.2f\n", conta.Saldo())
	}

	if err := conta.Sacar(1500); err != nil {
		fmt.Println("Erro ao sacar:", err)
	}

	fmt.Printf("Saldo final: R$%.2f\n", conta.Saldo())
}

File: 3/list-3.py
class ContaBancaria:
    def __init__(self, titular):
        self.__titular = titular  
        self.__saldo = 0.0        

    def depositar(self, valor):
        if valor > 0:
            self.__saldo += valor
            print(f"Depósito de R${valor:.2f} realizado com sucesso!")
        else:
            print("Valor de depósito deve ser maior que zero.")

    def sacar(self, valor):
        if valor > 0 and valor <= self.__saldo:
            self.__saldo -= valor
            print(f"Saque de R${valor:.2f} realizado com sucesso!")
        elif valor > self.__saldo:
            print("Saldo insuficiente para realizar o saque.")
        else:
            print("Valor de saque deve ser maior que zero.")

    def exibir_saldo(self):
        print(f"Saldo atual de {self.__titular}: R${self.__saldo:.2f}")
    
def main():

    conta = ContaBancaria("Marcio")
    conta.exibir_saldo()

    conta.depositar(100)
    conta.exibir_saldo()

    conta.sacar(30)
    conta.exibir_saldo()

    conta.sacar(80)
    conta.exibir_saldo()

if __name__ == "__main__":
    main()

